community_supplied/Sflow/check-mac-traffic.rule

```diff
@@ -0,0 +1,223 @@
+            }
+            field samplingRate {
+                sensor sflow-mac {
+                    path sampleSamplingRate;
+                }
+                type integer;
+                description "The rate at which packet is sampled.";
+            }
+            field sourceMac {
+                sensor sflow-mac {
+                    path recordSourceMac;
+                }
+                type string;
+                description "Source mac-address of the sampled packet.";
+            }
+            field totalBytes {
+                formula {
+                    eval {
+                        expression "$frame-length-mean * $samplingRate";
+                    }
+                }
+                type integer;
+                description "Product of the mean  frame size and sampling-rate.Gives the total traffic in bytes.";
+            }
+            /*
+             * Anomaly detection logic.
+             */			
+            trigger traffic {
+                frequency 60s;
+                term greaterthanhighthreshold {
+                    when {
+                        greater-than-or-equal-to "$totalBytes" "$highThreshold";
+                    }
+                    then {
+                        status {
+                            color red;
+                            message "On the Device $agentid , traffic of  $totalBytes Bytes for source address \"$sourceMac\" and destination address \"$destinationMac\" is above high threshold $highThreshold";
+                        }
+                    }
+                }
+                term greaterthanlowthreshold {
+                    when {
+                        greater-than-or-equal-to "$totalBytes" "$lowThreshold";
+                    }
+                    then {
+                        status {
+                            color yellow;
+                            message "On the Device $agentid , traffic of $totalBytes Bytes for source address \"$sourceMac\" and destination address \"$destinationMac\" is above low threshold $lowThreshold";
+                        }
+                    }
+                }
+                term normaltraffic {
+                    then {
+                        status {
+                            color green;
+                            message "On the Device $agentid , traffic of $totalBytes Bytes for source address \"$sourceMac\" and destination address \"$destinationMac\"is normal.";
+                        }
+                    }
+                }
+            }
+            /*
+             * Variables with default values. Default values can be changed
+             * while deploying playbook instance.
+             */			
+            variable highthreshold {
+                value 100000000;
+                description "High threshold for a Gig interface.Since total traffic is calculated in bytes, the interface speed of 800Mbps/8 is taken.";
```

Comment:
s/High threshold for a Gig interface.Since total traffic is calculated in bytes, the interface speed of 800Mbps/8 is taken./Enter high threshold in bytes as total traffic is calculated in bytes e.g. 800Mbps is 80% for a gigabit interface/g

Change in all rules for low and high thresholds